src/StaticArrays.jl

```diff
@@ -6,14 +6,21 @@ import Base: @_inline_meta, @_propagate_inbounds_meta, @_pure_meta, @propagate_i
 
 import Base: getindex, setindex!, size, similar, vec, show,
              length, convert, promote_op, promote_rule, map, map!, reduce, reducedim, mapreducedim,
-             mapreduce, broadcast, broadcast!, conj, transpose, ctranspose,
+             mapreduce, broadcast, broadcast!, conj, transpose,
              hcat, vcat, ones, zeros, eye, one, cross, vecdot, reshape, fill,
              fill!, det, logdet, inv, eig, eigvals, eigfact, expm, logm, sqrtm, lyap, trace, kron, diag, vecnorm, norm, dot, diagm, diag,
              lu, svd, svdvals, svdfact, factorize, ishermitian, issymmetric, isposdef,
              iszero, sum, diff, prod, count, any, all, minimum,
              maximum, extrema, mean, copy, rand, randn, randexp, rand!, randn!,
              randexp!, normalize, normalize!, read, read!, write, Eigen
 
+if VERSION >= v"0.7.0-DEV.1415"
+    import Base.adjoint
+else
+    import Base.ctranspose
+    const adjoint = ctranspose
```

Comment:
I assume there's nothing for this in `Compat` yet?

Reply:
There is. The alternative would be this:
```julia
import Compat.adjoint
export adjoint
```
and bump to `Compat 0.32` in `REQUIRE`. I thought this was a bit more confusing than the current implementation, since we essentially extend `Compat.adjoint` rather than the base method. Which solution do you prefer?

Reply:
On second thought that might bet better after all, since then users can use `adjoint` on v0.6 as well.

Reply:
Doesn't Compat deal with that anyway?

Reply:
After an explicit `using Compat`, (i.e. you would have to to `using StaticArrays; using Compat`).

Reply:
That is just fine (and transparent).

Reply:
Okay, so I'll remove the `export adjoint`?

Reply:
Yep, let's use Compat for this, and remove the export. Oh, and I see that you've already done this - nice.